uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar.component.js

```diff
@@ -266,59 +282,123 @@ define([
      */
     function monthStructure (dateMoment) {
       return {
-        index: dateMoment.month(),
+        index: getMonthIndex(dateMoment),
+        month: dateMoment.month(),
         year: dateMoment.year(),
         name: dateMoment.format('MMMM'),
-        shortName: dateMoment.format('MMM')
+        moment: moment().year(dateMoment.year()).month(dateMoment.month())
       };
     }
 
     /**
-     * Reloads the selected months data
+     * Paginates the currently selected month in a specified direction
+     *
+     * @param {String} direction previous|next
+     */
+    function paginateMonth (direction) {
+      var monthAction = direction === 'previous' ? 'subtract' : 'add';
+      var dateFromMonth = vm.selectedMonth.moment[monthAction](1, 'month');
+
+      vm.selectedMonthIndex = getMonthIndex(dateFromMonth);
+
+      setSelectedMonth();
+      setMonthPaginatorsAvailability();
+      refresh('month');
+    }
+
+    /**
+     * Reloads the selected month's data
      *
      * If the source of the refresh is a period change, then
      * it rebuilds the months list as well
      * If the source of the refresh is a change in contacts filters, then
      * it reloads the contacts as well
      *
-     * @param {string} source The source of the refresh (period or contacts change)
+     * @param {String} source The source of the refresh (period or contacts change)
      */
     function refresh (source) {
-      source = _.includes(['contacts', 'period'], source) ? source : 'period';
+      source = _.includes(['contacts', 'period', 'month'], source) ? source : 'period';
```

Comment:
```js
var resolveSource;
var sourceResolvers = {
  contacts: loadContacts,
  month: setMonthPaginatorsAvailability,
  period: setFirstPeriodMonth
};

source = source || 'period';
resolveSource = sourceResolvers[source];

$q.resolve()
  .then(makeSureMonthIsNotInjected)
  .then(resolveSource)
  .then(function () {
    injectAndShowMonth((source === 'contacts'));
  });
```

Reply:
There are a couple of issues here:
1) `vm.refresh` is public. `source = _.includes(...)` ensures that it always defaults to `period` even if smth unexpected is passed. better not to change that.
2) period executes 2 functions, not one.

So I'd prefer to leave this code unchanged.